src/common/utils/logger_helper.h

```diff
@@ -34,7 +34,15 @@ namespace LoggerHelpers
         std::filesystem::path logFolderPath(appPath);
         logFolderPath.append(LogSettings::logPath);
 
-        for (const auto& dir : std::filesystem::directory_iterator(logFolderPath))
+        std::error_code err;
+        auto folders = std::filesystem::directory_iterator(logFolderPath, err);
+        if (err.value())
+        {
+            Logger::error("Failed to create directory iterator for {}. {}", logFolderPath.string(), err.message());
```

Comment:
If the folder path is not present this is not an error that needs to be logged.

Reply:
Creating a folder if it is not present

Reply:
Returning immediately if the directory does not exist.